src/osd/ReplicatedPG.cc

```diff
@@ -11811,7 +11811,8 @@ void ReplicatedPG::agent_setup()
 void ReplicatedPG::agent_clear()
 {
   agent_stop();
-  agent_state.reset(NULL);
+  if (agent_state)
+    agent_state.reset(NULL);
```

Comment:
boost scope_ptr always reset the inner by calling a swap (this_type(p).swap(*this);) as well as some sanity checks. Therefore, if it is already null, the clear process is not needed